### PR TITLE
Fix issues when installing Redis

### PR DESCRIPTION
**Pull request by remotedevops**
We need to create redis.conf and redis-server.service before installing the redis package.

### CODE REVIEW

**Comment by dvalfre** [comment]
Tested the modifications through an overlay.  Works without issues.

File: cookbooks/redis/recipes/install.rb [hunk 1]
```diff
@@ -17,6 +17,12 @@
 
 run_installer = !FileTest.exists?(redis_base_directory) || node['redis']['force_upgrade']
 
+# check if redis-server exists
```
Comment (jfuechsl) [inline cookbooks/redis/recipes/install.rb]: Move the block in https://github.com/engineyard/ey-cookbooks-stable-v6/pull/100/files#diff-282ec2fdc23e486143f8455f9feab8d2R104 up and use the `redis_bin_path` in the checks here.

File: cookbooks/redis/recipes/install.rb [hunk 2]
```diff
@@ -114,6 +112,10 @@
     action :nothing
   end
 
+  # check if redis is installed
+  # if it is, restart it when redis-server.service changes
+  # without the check, "notifies :restart" fails when redis-server isn't installed yet
+  is_redis_installed = Mixlib::ShellOut.new("[ -e #{redis_bin_path} ]").run_command.exitstatus == 0
```
Comment (jfuechsl) [inline cookbooks/redis/recipes/install.rb]: Why can't we use `run_installer`?

File: cookbooks/redis/recipes/install.rb (hunk 1 above)
Comment (remotedevops) [reply cookbooks/redis/recipes/install.rb]: Made the change.

File: cookbooks/redis/recipes/install.rb (hunk 2 above)
Comment (remotedevops) [reply cookbooks/redis/recipes/install.rb]: If redis isn't installed (during the first time the recipe runs), you can't call `notifies :restart, "service[redis-server]", :immediately`. Maybe I can remove `:immediately`?

Comment (remotedevops) [reply cookbooks/redis/recipes/install.rb]: I removed :immediately and removed is_redis_installed.

**Pull request merged**